hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/ContractResultServiceImpl.java

```diff
@@ -384,19 +392,23 @@ private void processCreatedContractEntity(RecordItem recordItem, EntityId contra
     }
 
     @SuppressWarnings("java:S3776")
-    private ByteString processSidecarRecords(RecordItem recordItem) {
-        ByteString failedInitcode = null;
-        var sidecarRecords = recordItem.getSidecarRecords();
+    private SidecarProcessingResult processSidecarRecords(
+            final RecordItem recordItem, final TransactionBody transactionBody) {
+        final var sidecarRecords = recordItem.getSidecarRecords();
         if (sidecarRecords.isEmpty()) {
-            return failedInitcode;
+            return new SidecarProcessingResult(null, false, null);
         }
 
         var contractBytecodes = new ArrayList<ContractBytecode>();
         int migrationCount = 0;
         var stopwatch = Stopwatch.createStarted();
 
-        for (var sidecarRecord : sidecarRecords) {
-            boolean migration = sidecarRecord.getMigration();
+        long topLevelActionSidecarGasUsed = 0;
```

Comment:
nit

```suggestion
        Long topLevelActionSidecarGasUsed = null;
```

then the ternary expression at the end is not needed

Reply:
Good suggestion. Applied it.